### PR TITLE
Add `notifications_enabled` workflow to disable nofications

### PR DESCRIPTION
**Pull request by devnoname120**

### CODE REVIEW

File: TemporaryEmail/source/info.plist [hunk 1]
```diff
@@ -110,6 +112,7 @@ readonly front_browser="$(osascript -e 'tell application "System Events" to retu
 
 if [[ "${front_browser}" == 'Safari'* || "${front_browser}" == 'Webkit'* ]]; then
   osascript &lt;&lt;EOF
+    tell application "System Events" to keystroke "${email}"
```
Comment (devnoname120) [inline TemporaryEmail/source/info.plist:115]: Useful for me since I usually open the workflow when I'm in an input where I don't want to put my real email.

File: TemporaryEmail/source/info.plist [hunk 2]
```diff
@@ -118,14 +121,16 @@ if [[ "${front_browser}" == 'Safari'* || "${front_browser}" == 'Webkit'* ]]; the
 EOF
 elif [[ "${front_browser}" == 'Google Chrome'* || "${front_browser}" == 'Chromium'* || "${front_browser}" == 'Opera'* || "${front_browser}" == 'Vivaldi'* || "${front_browser}" == 'Brave Browser'* || "${front_browser}" == 'Microsoft Edge'* ]]; then
   osascript &lt;&lt;EOF
+    tell application "System Events" to keystroke "${email}"
     tell application "${front_browser}"
       set tabIndex to active tab index of front window
-      open location "${url}"
+      make new tab at after tab tabIndex of front window with properties {URL:"${url}"}
```
Comment (devnoname120) [inline TemporaryEmail/source/info.plist:127]: This has two upsides:
- The tab opens next to the current tab instead of at the end.
- It returns faster so the command `set active tab` is run earlier so the transition can't be noticed (it switches back before the tab is actually open).

File: TemporaryEmail/source/info.plist [hunk 3]
```diff
@@ -367,11 +375,13 @@ Because [Firefox does not support AppleScript](https://bugzilla.mozilla.org/show
 	</dict>
 	<key>variables</key>
 	<dict>
+		<key>notifications_enabled</key>
+		<string>true</string>
```
Comment (devnoname120) [inline TemporaryEmail/source/info.plist:379]: I set it to `true` by default but I use `false` for me.

File: TemporaryEmail/source/info.plist [hunk 4]
```diff
@@ -90,7 +90,9 @@
 				<integer>102</integer>
 				<key>script</key>
 				<string>function notification {
-  ./notificator --message "${1}" --title "${alfred_workflow_name}"
+  if [[ "${notifications_enabled}" == 'true' ]]; then
```
Comment (devnoname120) [inline TemporaryEmail/source/info.plist:93]: The notification is pretty annoying for me and useless. Also macOS asks me to enable notifications for `Notificator` which is annoying.

**Comment by devnoname120** [comment]
Request review from @vitorgalvao.

File: TemporaryEmail/source/info.plist (hunk 1 above)
Comment (vitorgalvao) [reply TemporaryEmail/source/info.plist:115]: I can see the usefulness, but I don’t agree with the implementation. Typing via AppleScript is finicky. It should instead output the email to a [Copy to Clipboard Output](https://www.alfredapp.com/help/workflows/outputs/copy-to-clipboard/) which can also paste it to the frontmost app. I’m making the change locally now.

File: TemporaryEmail/source/info.plist (hunk 4 above)
Comment (vitorgalvao) [reply TemporaryEmail/source/info.plist:93]: If you don’t want notifications, refuse them when macOS asks. There’s no reason to add more code and configuration for something macOS deals with.

File: TemporaryEmail/source/info.plist (hunk 3 above)
Comment (vitorgalvao) [reply TemporaryEmail/source/info.plist:379]: I appreciate that, even though this feature won’t be accepted.

File: TemporaryEmail/source/info.plist (hunk 2 above)
Comment (vitorgalvao) [reply TemporaryEmail/source/info.plist:127]: > It returns faster so the command set active tab is run earlier so the transition can't be noticed (it switches back before the tab is actually open).

I don’t see it. I tried it multiple times and it seemed about the same speed. I could definitely still notice the transition.

> The tab opens next to the current tab instead of at the end.

I can see the advantage and I’m up for adding it, but only if it’s added to the Safari-based browsers section as well. Bonus if there’s truly no transition, adding a new tab without switching to it.

Comment (devnoname120) [reply TemporaryEmail/source/info.plist:127]: I suppose it depends on configurations then. On my Macbook Air M1 (2020) running macOS Monterey 12.3.1 and Chrome 100.0.4896.60 (arm64) I could reliably reproduce the difference in speed. My intuition is that `open` goes through some system stuff whereas `make new tab` is a direct Chromium API, so it should bypass some abstraction layers.

I don't use Safari, would you be OK to implement these changes on your own? I enabled writing access to my branch for this repo's maintainers.

File: TemporaryEmail/source/info.plist (hunk 1 above)
Comment (devnoname120) [reply TemporaryEmail/source/info.plist:115]: I disagree. Typing via AppleScript works great as long as you're not using special characters such as emojis. The generated email addresses never contain anything that wouldn't work. Copying to clipboard is a pain because Alfred doesn't give provide any good way to restore the clipboard. So you have to do it manually by storing the previous clipboard in a env variable, and then restoring it via e.g. another Alfred action, Apple script or some shell commands. I'm concerned it won't play well with some clipboard managers, and it seems like a more fragile solution prone to race conditions (if the clipboard is modified between the time you save it and the time your restore it).

File: TemporaryEmail/source/info.plist (hunk 4 above)
Comment (devnoname120) [reply TemporaryEmail/source/info.plist:93]: Hmm good point, I remember it messed up another workflow but I'm not entirely sure why anymore… I'll try and report back.

File: TemporaryEmail/source/info.plist (hunk 1 above)
Comment (vitorgalvao) [reply TemporaryEmail/source/info.plist:115]: > works great as long as

I’ve seen AppleScript typing screw up in uppercases. I wouldn’t be surprised if it screwed up on `@`.

> Alfred doesn't give provide any good way to restore the clipboard.

Tell it to copy `{clipboard:1}`. Done.

Regardless of your feelings about copying to the clipboard, that’s what the Workflow was doing already and you didn’t remove that part. All I’m doing is moving the copy logic to another object so it can be pasted at the same time. Plus, I want the email in clipboard. Often I have to reference it again a short time after, so having it in the history is a feature.

File: TemporaryEmail/source/info.plist (hunk 2 above)
Comment (vitorgalvao) [reply TemporaryEmail/source/info.plist:127]: > I don't use Safari

I don’t use all browsers I support in my Workflows, but I support them nonetheless. Barring technical limitations, if multiple browsers are added then features should be the equivalent across them or not be added at all. I don’t like things half implemented.

> would you be OK to implement these changes on your own?

Yes, but if I’m doing it myself might as well close the PR since I’m literally only going to use one line from it. No promises as to when I’ll look into it, though. Especially because if I do it, I’ll again look into the possibility of making the transition truly seamless.

Thank you anyway.

Comment (devnoname120) [reply TemporaryEmail/source/info.plist:127]: Alright, sounds good to me. I don't really agree about your clipboard comments and Applescript's keystrokes, but I'm not sure it's worth arguing more over. Also it's too minor to warrant a hard fork on my side.

Small feedback regarding the reviewing style because I think people rarely muster up the courage but it's important nonetheless.
- It was pretty disagreeable to read your comments because you said unnecessary stuff such as ‶regardless of your feelings″, which is pretty dismissive and sounds to be based on bad faith. I argumented rationally so reframing it as ‶feelings″ is false and pretty annoying.
- Another one was ‶I don’t like things half implemented″, which is dismissive as well (it implies that my PR is ‶half implemented″). If you were to ask me, I'd suggest to rephrase it to something like ‶I personally prefer ensuring consistency of features across browsers rather than providing the best experience for each browser given the APIs that they each provide″.
- Also, saying ‶don't like″ sounds like it's an arbitrary choice based on feelings, rather than a careful assessment of pros and cons. This is at odds with the alleged dismissal of feelings; you can't just dismiss (what you identify as) people's feelings and then voice yours.

Hope this feedback won't rub you the wrong way, that's not the intent. Thank you for this workflow though it's pretty great.

Comment (vitorgalvao) [reply TemporaryEmail/source/info.plist:127]: > Small feedback

Thank you for the feedback. Truly. I too think it’s important.

> you said unnecessary stuff such as ‶regardless of your feelings″

It was meant as “regardless of your preference”. In the sense that I have other users to think about (including myself) so a change in behaviour should consider how it affects everyone and not a single person. Was it unnecessary? Arguably. Could I have communicated it differently? Your feedback is clear on “yes”. Was it dismissive? I would not say so, as that was not what I intended to convey.

> it implies that my PR is ‶half implemented″

I didn’t make an implication, but I do see how it could be read that way. In Portugal we’d call that “mandar uma boca” (literally “throw a mouth”) and I abhor it. I do apologise for the ambiguity; I could’ve cut that sentence and the point would have retained its meaning. Had I realised while writing that it could be misinterpreted, I would have left it out.

To clarify, it was written as an explanation of my methodology, not a comment on yours. More importantly, *a comment on your code would not have been a comment on yourself*. If anything, I found your PR to be respectful and comprehensive.

> I'd suggest to rephrase it to (…) which would be both more accurate and more diplomatic.

It’s also four times longer and we both have other things to do. Diplomacy goes both ways: you say things to me in good faith and I in turn assume you’re saying things in good faith. When we both do that, we expedite our points and reach a conclusion faster.

> ensuring consistency of features across browsers rather than providing the best experience for each browser

I think you’d like me to address this, because it is incorrect. “Barring technical limitations” is the operative sentence. The goal is to provide consistency of features *when it’s technically feasible*, which in this case we knew it was. But if a browser doesn’t support something I don’t “punish” the others.

> saying ‶don't like″ sounds like it's an arbitrary choice based on feelings (…) no, it's not a universal rule.

Precisely. It is because it’s not a universal rule that I used “don’t like”. Everything in software development has a tradeoff and some things are subjective. I don’t *have to* support every browser, I *choose* to. Are you wrong for thinking differently? No, not at all! But code that is merged into my repo is code I have to support, so it should be code I am happy with.

[I believe in empathy, not animosity or rudeness](https://www.alfredforum.com/topic/10576-power-thesaurus-search/page/2/#comment-93948), and in [seeing things in the best light](https://youtu.be/eC7xzavzEKY), so I do hope you believe me when I say any negatives feelings I caused were unintended.

Thank you again for the feedback and the PR. And the nice words on the Workflow. Have a great week!

File: TemporaryEmail/source/info.plist (hunk 1 above)
Comment (devnoname120) [reply TemporaryEmail/source/info.plist:115]: @vitorgalvao Need to use `{clipboard:0}` instead if the item is marked as transient.

In addition it doesn't work with non-textual clipboard content e.g. images.
